src/main/java/com/qubit/solution/fenixedu/integration/cgd/domain/logs/CgdCommunicationLog.java

```diff
@@ -0,0 +1,45 @@
+package com.qubit.solution.fenixedu.integration.cgd.domain.logs;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.fenixedu.academic.domain.Person;
+import org.fenixedu.academic.domain.student.Registration;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.joda.time.DateTime;
+
+public class CgdCommunicationLog extends CgdCommunicationLog_Base {
+
+    public CgdCommunicationLog(Registration registration, boolean requestCard, boolean success, Person sender) {
+        setRegistration(registration);
+        setRequestCard(requestCard);
+        setSuccess(success);
+        setSender(sender);
+        setSendDate(DateTime.now());
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    public static Collection<CgdCommunicationLog> findLogsByStudent(Person person) {
+        return Bennu.getInstance().getCgdComunicationLogsSet().stream()
+                .filter(log -> log.getRegistration().getPerson().equals(person))
+                .sorted((log1, log2) -> log2.getSendDate().compareTo(log1.getSendDate())).collect(Collectors.toSet());
+    }
+
+    public static Optional<CgdCommunicationLog> getStudentLatestCgdCommunicationLog(Person person) {
```

Comment:
@miguel-morais Usually we use find in these methods, you even have findStudentLogs. So I suggest the API is:

- Collection<CgdCommunicationLog> findStudentsLogs
- Optional<CgdCommunicationLog> findLatestStudentLog